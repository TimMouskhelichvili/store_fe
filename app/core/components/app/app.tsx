import 'core/components/app/app.scss';

import * as React from 'react';
import { ApolloProvider } from 'react-apollo';
import { Direction } from 'core/components/sidemenu/enums/direction';
import { Header } from 'core/components/header/header';
import { Sidemenu } from 'core/components/sidemenu/sidemenu';
import { client } from 'core/apollo/client';

class App extends React.Component {
    public render(): React.ReactNode {
        return (
            <ApolloProvider client={client}>
                <Header />
                <Sidemenu align={Direction.LEFT} variable='sidemenu' />
                <Sidemenu align={Direction.RIGHT} variable='options' />
            </ApolloProvider>
        );
    }
}

export {
    App
};
