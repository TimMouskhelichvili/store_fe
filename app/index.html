<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=0">
    <link rel="profile" href="http://gmpg.org/xfn/11">
    <title>Store v2</title>
    <base href="/">
</head>

<body>
    <div id="index"></div>
</body>

</html>