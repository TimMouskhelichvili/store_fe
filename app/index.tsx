import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { App } from 'core/components/app/app';

ReactDOM.render(<App />, document.getElementById('index'));