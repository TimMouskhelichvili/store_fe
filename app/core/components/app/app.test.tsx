import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { App } from 'core/components/app/app';

describe('Core component app tests', (): void => {
    it('renders', (): void => {
        const component = renderer.create(<App />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});
