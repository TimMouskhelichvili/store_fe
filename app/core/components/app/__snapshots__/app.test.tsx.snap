// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core component app tests renders 1`] = `
Array [
  <header
    className="header"
  >
    <button
      id="sidemenu_button"
      onClick={[Function]}
    >
      sidemenu
    </button>
    <button
      id="options_button"
      onClick={[Function]}
    >
      options
    </button>
  </header>,
  <div
    className="container"
    id="sidemenu_sidemenu_container"
  />,
  <div
    className="opacity"
    id="sidemenu_sidemenu_opacity"
    onClick={[Function]}
    onTransitionEnd={[Function]}
  />,
  <div
    className="container"
    id="options_sidemenu_container"
  />,
  <div
    className="opacity"
    id="options_sidemenu_opacity"
    onClick={[Function]}
    onTransitionEnd={[Function]}
  />,
]
`;
